### DISCUSSION TITLE
The database log file is too large to clean up （sqlserver）

### DISCUSSION THREAD

**Discussion by ourlang** [nextgenhealthcare/connect#6024] [Q&A]
1、Connect to the database and configure it as SQL Server
2、The validity period of the log is set to 2 days
![image](https://github.com/nextgenhealthcare/connect/assets/32355185/a5254087-92d7-41d1-afad-1581f4c112ee)
3、There are three tables in the database that take up a lot of space and cannot be cleaned up
D_MC1 、D_MM1,D_M1
![image](https://github.com/nextgenhealthcare/connect/assets/32355185/e1819463-3c29-4ffe-9de9-6358914ff9a8)
![image](https://github.com/nextgenhealthcare/connect/assets/32355185/f0790e7d-7a80-43fd-ab34-6f86a51616b8)
<img width="548" alt="1702437873998" src="https://github.com/nextgenhealthcare/connect/assets/32355185/9c2f7fbe-13b3-4cf0-9423-0892803eca69">

4、How to make scheduled cleaning of logs and database files effective

**Comment by pacmano1** [comment]
Most folks prune every hour.  Your frequency likely can't keep up.

**Comment by ourlang** [comment]
In addition to setting up here data prune , please  Is there any other way to reduce the disk space occupied by large database files or to regularly clean up database files？

**Comment by ourlang** [comment]
The database transaction files generated every day occupy 5G and require manual processing each time. Do you have any other configurations? Thank you

**Comment by pacmano1** [comment]
That is a SQL server exercise, not a Mirth question.   Set recovery mode to simple.   I assume you have a SQL job to automate all the SQL server stuff like truncating the log on backup.  If not, you should use the maintenance solution at https://ola.hallengren.com/.   Best stuff ever.

**Comment by pacmano1** [comment]
I suspect your channel may also be poorly designed and not use attachment handlers.

**Reply by ourlang** [reply]
what is attachment handlers

**Reply by ourlang** [reply]
![image](https://github.com/user-attachments/assets/97f49d24-2778-4f46-acfe-7db427a42a69)

Is this configuration correct？

**Reply by pacmano1** [reply]
Please go read up on attachments if you are processing large files.    https://forums.mirthproject.io/articles/175046-using-attachments.

Those settings are fine - but you also need to check that pruning is enabled under settings.

**Comment by pacmano1** [comment]
Please stop posting to issues unless there is a known bug or you have a feature request.  I am converting this to a discussion.
